command.go

```diff
@@ -117,16 +117,16 @@ func (c *Command) Close() error {
 		return err
 	}
 
-	select { // close cancel channel if it's still open
-	case <-c.cancel:
-	default:
-		close(c.cancel)
-	}
+	close(c.cancel)
```

Comment:
I think the `select` here was made to prevent crashing when calling `Close` twice.
I believe this should instead be protected by a `sync.Once` instead.
Leaving the `close(c.cancel)` as is, makes it at risk of double close and panic.

Reply:
Sync.once around the close seems like a workaround without fixing the root cause, but it may be that a mutex is going to be needed somewhere, perhaps to guard access to `c.id`.